### PR TITLE
Damage animation blending via 2DA

### PR DESCRIPTION
**Pull request by MarcelHB**
## Description
See #2050, and also #2046 as a side node.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [x] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: gemrb/unhardcoded/bg2ee/damage.2da [hunk 1]
```diff
@@ -1,16 +1,16 @@
 2DA V1.0
 *
 	MAIN		SPARKS		GRADIENT
```
Comment (bradallred) [inline gemrb/unhardcoded/bg2ee/damage.2da:3]: missing column header?

Comment (bradallred) [reply gemrb/unhardcoded/bg2ee/damage.2da:3]: looks like a few of these files have that issue

File: gemrb/core/Scriptable/Actor.cpp [hunk 2]
```diff
@@ -2371,44 +2380,52 @@ void Actor::PlayDamageAnimation(int type, bool hit)
...
 			}
 			break;
 		case 13: case 14: case 15://acid
 			if (hit) {
-				AddAnimation(d_main[type], d_gradient[type], height, flags);
+				AddAnimation(damageMainResources[type], damageGradients[type], height, flags);
+			}
+
+			for (i = DL_ACID; i <= type; i++) {
+				AddAnimation(damageSparks[i], damageGradients[i], height, flags);
```
Comment (lynxlynxlynx) [inline gemrb/core/Scriptable/Actor.cpp:2423]: nice catch!

File: gemrb/core/Scriptable/Actor.cpp [hunk 3]
```diff
@@ -2352,13 +2358,16 @@ void Actor::PlayDamageAnimation(int type, bool hit)
 {
 	if (!anims) return;
 	int i;
-	int flags = AA_PLAYONCE;
+	int flags = AA_PLAYONCE|AA_BLEND;
```
[the remark this answers is not in the document]
Comment (MarcelHB) [reply gemrb/core/Scriptable/Actor.cpp]: Debugging leftover ...

[the remark this answers is not in the document]
Comment (lynxlynxlynx) [reply gemrb/core/Scriptable/Actor.cpp]: I hope it doesn't invalidate all the table data ...

File: gemrb/unhardcoded/bg2ee/damage.2da (hunk 1 above)
Comment (lynxlynxlynx) [reply gemrb/unhardcoded/bg2ee/damage.2da]: still missing for HoW

Comment (MarcelHB) [reply gemrb/unhardcoded/bg2ee/damage.2da]: Thanks, missed beautifing some of the research bits.

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**